thirdparty/github.go

```diff
@@ -284,57 +335,20 @@ func GetGithubCommits(ctx context.Context, token, owner, repo, ref string, until
 		options.Until = until
 	}
 
-	installationToken, err := GetInstallationToken(ctx, owner, repo, nil)
-	if err != nil {
-		ctx, httpClient, putClient := getGithubClient(ctx, installationToken, "GetGithubCommits", retryConfig{retry: true}, []attribute.KeyValue{
-			attribute.String(githubOwnerAttribute, owner),
-			attribute.String(githubRepoAttribute, repo),
-			attribute.String(githubRefAttribute, ref),
-		})
-		defer putClient()
-		client := github.NewClient(httpClient)
-
-		commits, resp, err := client.Repositories.ListCommits(ctx, owner, repo, &options)
-		if resp != nil {
-			defer resp.Body.Close()
-			if err == nil {
-				return commits, resp.NextPage, nil
-			}
-		}
-
-		grip.Debug(message.WrapError(err, message.Fields{
-			"ticket":  "EVG-19966",
-			"message": "failed to get commits from GitHub",
-			"caller":  "GetGithubCommits",
-			"owner":   owner,
-			"repo":    repo,
-			"options": options,
-			"resp":    resp,
-		}))
-	}
-
-	// Fallback to not using the GitHub app on error.
-	ctx, legacyClient, putLegacyClient := getGithubClient(ctx, token, "GetGithubCommits", retryConfig{retry: true}, []attribute.KeyValue{
-		attribute.String(githubOwnerAttribute, owner),
-		attribute.String(githubRepoAttribute, repo),
-		attribute.String(githubRefAttribute, ref),
-	})
-	defer putLegacyClient()
-	client := github.NewClient(legacyClient)
-
-	commits, newResp, err := client.Repositories.ListCommits(ctx, owner, repo, &options)
-	if newResp != nil {
-		defer newResp.Body.Close()
+	commits, resp, err := githubClient.Repositories.ListCommits(ctx, owner, repo, &options)
+	if resp != nil {
+		defer resp.Body.Close()
+		span.SetAttributes(attribute.Bool(githubCachedAttribute, respFromCache(resp.Response)))
```

Comment:
what's this bit about?

Reply:
This is to let us query the ratio of requests that get their answer from the cache (so they don't count towards the rate limit) to the requests that need to get an original response from GitHub. This will help us to gauge how effective the cache is.